javacutil/src/main/java/org/checkerframework/javacutil/TreeUtils.java

```diff
@@ -247,6 +247,32 @@ public static boolean isSelfAccess(ExpressionTree tree) {
     return false;
   }
 
+  /**
+   * Returns true iff the underlying JCTree exists and start at a position p &le; 0 in the source
+   * code.
+   *
+   * <p>Note that there are trees that may actually start at position 0 in the source code, such as
+   * import statements or class declarations. Be careful.
```

Comment:
While "be careful" is good advice generally, it's worrying non-specific here. What would be the consequences if this method were to be called improperly? I would phrase this as under-approximation: this method does not return true _iff_ the tree is synthetic: it returns true _if_ the tree is synthetic (and also for some other trees, too).